utils/strip-metadata/src/lib.rs

```diff
@@ -0,0 +1,946 @@
+                method_inputs.chain(core::iter::once(method_output))
+            });
+
+        // The extrinsic type IDs:
+        let extrinsic_type_ids = [
+            &mut self.extrinsic.address_ty.id,
+            &mut self.extrinsic.signature_ty.id,
+        ];
+
+        // Outer enum type IDs:
+        let outer_enum_type_ids = [
+            &mut self.outer_enums.call_enum_ty.id,
+            &mut self.outer_enums.event_enum_ty.id,
+            &mut self.outer_enums.error_enum_ty.id,
+        ];
+
+        // Transaction Extension types:
+        let transaction_extension_types = self
+            .extrinsic
+            .transaction_extensions
+            .iter_mut()
+            .flat_map(|ext| [&mut ext.ty.id, &mut ext.implicit.id].into_iter());
+
+        // Custom types:
+        let custom_type_ids = self.custom.map.values_mut().map(|value| &mut value.ty.id);
+
+        // Return all IDs gathered:
+        pallet_types
+            .chain(runtime_api_types)
+            .chain(extrinsic_type_ids)
+            .chain(outer_enum_type_ids)
+            .chain(transaction_extension_types)
+            .chain(custom_type_ids)
+    }
+}
+
+/// This trait defines how to get a type registry from the metadata
+trait GetTypes {
+    fn get_types_mut(&mut self) -> &mut PortableRegistry;
+}
+
+impl GetTypes for v14::RuntimeMetadataV14 {
+    fn get_types_mut(&mut self) -> &mut PortableRegistry {
+        &mut self.types
+    }
+}
+
+impl GetTypes for v15::RuntimeMetadataV15 {
+    fn get_types_mut(&mut self) -> &mut PortableRegistry {
+        &mut self.types
+    }
+}
+
+impl GetTypes for v16::RuntimeMetadataV16 {
+    fn get_types_mut(&mut self) -> &mut PortableRegistry {
+        &mut self.types
+    }
+}
+
+/// Subxt needs this type so we always ensure to preserve it
+/// even if it's not explicitly mentioned anywhere:
+fn find_dispatch_error_type(types: &mut PortableRegistry) -> u32 {
+    types
+        .types
+        .iter()
+        .enumerate()
+        .find(|(_idx, ty)| ty.ty.path.segments == ["sp_runtime", "DispatchError"])
+        .expect("Metadata must contain sp_runtime::DispatchError")
+        .0 as u32
+}
+
+/// Create Either enums which can be iterated over.
+macro_rules! either{
+    ($name:ident: $first_tok:ident $($tok:ident)*) => {
+        enum $name<$first_tok, $($tok),+> {
+            $first_tok($first_tok),
+            $( $tok($tok), )*
+        }
+
+        impl <$first_tok, $($tok),+> Iterator for $name<$first_tok, $($tok),+>
+        where
+            $first_tok: Iterator,
+            $($tok: Iterator<Item = $first_tok::Item>,)*
+        {
+            type Item = $first_tok::Item;
+            fn next(&mut self) -> Option<Self::Item> {
+                match self {
+                    $name::$first_tok(item) => item.next(),
+                    $($name::$tok(item) => item.next(),)*
+                }
+            }
+        }
+    }
+}
+
+either!(Either: A B);
```

Comment:
maybe more conventional `Left` and `Right` , also why not use `either` that is already a dependency? is it because of finnicky `Iter` trait defs on it?

Reply:
Good point! This is a bit of a vestige from when I had a couple of Either defs (which is also why not `Left` and `Right`, because I had an `Either6` with 6 branches, so the `Either` aligned with using letters instead!)

I deleted this and justu sed the `either ` crate!